R/readWriteMgfData.R

```diff
@@ -45,13 +60,17 @@ writeMgfDataFile <- function(splist, con, COM = NULL, TITLE = NULL,
     if (verbose)
       setTxtProgressBar(pb, i)
 
-    writeMgfContent(splist[[i]], TITLE = NULL, con = con)
+    if (haveAddFields)
```

Comment:
Do we really need this `if` statement here. If we reach this point `addFields` is a `matrix` or `NULL`. And you can subset `NULL`:

```r
f <- function(n, addFields=NULL) {
    for (i in seq_len(n)) {
        print(addFields[i, ])
    }
}

f(3)
# NULL
# NULL
# NULL

f(3, matrix(1:6, ncol=2, dimnames=list(NULL, c("FOO", "BAR"))))
# FOO BAR 
#   1   4 
# FOO BAR 
#   2   5 
# FOO BAR 
#   3   6 
```

While it looks a little bit odd/dirty the workhorse function `writeMgfContent` will test for `NULL` itself.

Reply:
👍